File: finger_exercises/chapter3/prime_divisor.py
"""Test if an int x > 2 is prime. If not, print the largest divisor of x"""

x = int(input("Enter an integer greater than 2: "))
largest_divisor = None
if x % 2 == 0:
    largest_divisor = x / 2
for guess in range(3, x, 2):
    #print(guess)
    if x % guess == 0:
        largest_divisor = x / guess
        break
if largest_divisor != None:
    print(f"Largest divisor of {x} is {largest_divisor:.0f}.")
else:
    print(f"{x} is prime.")

File: finger_exercises/chapter3/prime_divisor.c
#include <stdio.h>

int main(void){
    int x;
    int largest_divisor = 1;
    int guess = 3;
    int result;

    printf("Enter an integer greater than 2: ");
    scanf("%d", &x);

    if (x % 2 == 0){
        largest_divisor = x / 2;
    }
    else{
        while (guess < x)
    {
        result = x % guess;
        if (result == 0)
        {
            largest_divisor = x / guess;
            //printf("largest_divisor is %d\n", largest_divisor);
            break;
        }
        guess++;
    }

    }
    
    if (largest_divisor == 1)
    {
        printf("%d is prime\n", x);
    }
    else{
        printf("Largest divisior is %d\n", largest_divisor);

    return 0;
    
    }
}